transcript/src/main/java/com/transcriptanalyzer/transcript/Tree.java

```diff
@@ -71,17 +78,95 @@ public Node hasChild(String intent){
+                    // take the best leaf data of the current child
+                    LinkedHashMap<String, Integer> childBestLeaves = child.getBestLeafData();
+
+                    // for each entry of this child's best leaf data, merge the entry into leafData
+                    for (Map.Entry<String, Integer> leaf: childBestLeaves.entrySet()) {
+                        // get the intent and occurrence of the current entry
+                        String intent = leaf.getKey();
+                        Integer occurrences = leaf.getValue();
+
+                        // merge it into leaf data. If an occurrence is new, it is added to the LinkedHashMap.
+                        // If it already exists, their values are summed using Integer::sum
+                        leafData.merge(intent, occurrences, Integer::sum);
+                    }
+                }
+
+                // move the leaves into a list
+                List<Map.Entry<String, Integer>> leafDataEntries = new ArrayList<>(leafData.entrySet());
+
+                // sort the list using the Map.Entry comparing by value comparator in reverse order
+                // this sorts each hash map entry, which is stored in a list, in descending order
+                leafDataEntries.sort(Map.Entry.<String, Integer>comparingByValue().reversed());
+
+                // determine how many entries to take from the list
+                // take the last index of the entries we will take. no more than 3,
+                // and no more than the number of elements in the list
+                int last_index = java.lang.Math.min(2, leafDataEntries.size() - 1);
+
+                // for each index of the entries we want
+                for (int i = 0; i <= last_index; i++) {
+                    // directly extract the key and value at that index
+                    String key = leafDataEntries.get(i).getKey();
+                    Integer value = leafDataEntries.get(i).getValue();
+
+                    // put them into the new sorted map. Has 0 to 3 values in it. In the context of our problem,
+                    // should have 1 to 3.
+                    // we can put instead of merging without overwriting because we know the entries in leafDataEntries
+                    // are unique; they were created by merging entries.
+                    bestLeafData.put(key, value);
+                }
+            }
+            return bestLeafData;
+        }
     }
 
     // get a list of the three nodes with the greatest occurrences sorted in decreasing order
-    public List<String> getTopThree(){
+    public List<String> getBestTreeIntents(){
         // compares each value and get the top 3, return key list attributed to these top 3
         return this.counts.entrySet().stream().sorted(
                 Map.Entry.<String, Integer>comparingByValue().reversed()).limit(3).map(
                         Map.Entry::getKey).collect(Collectors.toList());
     }
```

Comment:
We should really consider dividing this up somehow as it's rather deeply nested. Good job coming up with this however haha!